src/clock-weather-card.ts

```diff
@@ -336,8 +338,10 @@ export class ClockWeatherCard extends LitElement {
 
   private getWeather(): Weather {
     const weather = this.hass.states[this.config.entity] as Weather | undefined;
+    const temp = this.hass.states[this.config.temperature_sensor] as TemperatureSensor | undefined;
     if (!weather) throw new Error('Weather entity could not be found.');
     if (!weather?.attributes?.forecast) throw new Error('Weather entity does not have attribute "forecast".');
+    if (temp) weather.attributes.temperature = +temp.state;
```

Comment:
please do not manipulate the global weather, but create a method `getCurrentTemperature()` where you either read the value from `config.temperature_sensor`, if defined, otherwise from `config.entity`

Reply:
added funtion getLocalizedTemperature(), which returns temperature and unit as configured in the system settings.